libs/sdk-ui-kit/src/RecurrenceForm/useCronValidation.ts

```diff
@@ -0,0 +1,146 @@
+            const trimmedCronExpression = expression.replace(/\s+/g, "");
+            const parts = expression.split(/\s+/).filter(Boolean);
+
+            if (trimmedCronExpression.length === 0) {
+                return RecurrenceFormCronErrorTypes.EMPTY;
+            }
+
+            const isValid = isCronExpressionValid(expression, allowHourlyRecurrence);
+            if (!isValid) {
+                return RecurrenceFormCronErrorTypes.FREQUENCY_TOO_HIGH;
+            }
+
+            if (parts.length !== 6) {
+                return RecurrenceFormCronErrorTypes.WRONG_FORMAT;
+            }
+
+            const isValidCron = parseCron(expression);
+            if (!isValidCron) {
+                return RecurrenceFormCronErrorTypes.WRONG_FORMAT;
+            }
+
+            return null;
+        },
+        [allowHourlyRecurrence, parseCron],
+    );
+
+    const handleValidation = useCallback(
+        (expression: string) => {
+            const validationResult = validate(expression);
+
+            if (validationResult) {
+                switch (validationResult) {
+                    case RecurrenceFormCronErrorTypes.EMPTY:
+                        setCronError(errorMessages.empty.id);
+                        break;
+                    case RecurrenceFormCronErrorTypes.WRONG_FORMAT:
+                        setCronError(errorMessages.wrongFormat.id);
+                        break;
+                    case RecurrenceFormCronErrorTypes.FREQUENCY_TOO_HIGH:
+                        setCronError(errorMessages.frequencyTooHigh.id);
+                        break;
+                }
+            } else {
+                setCronError(null);
+            }
+
+            return !validationResult;
+        },
+        [validate],
+    );
+
+    const handleChange = useCallback(
+        (expression: string) => {
+            const validationResult = validate(expression);
+
+            if (cronError) {
+                if (validationResult) {
+                    switch (validationResult) {
+                        case RecurrenceFormCronErrorTypes.EMPTY:
+                            setCronError(errorMessages.empty.id);
+                            break;
+                        case RecurrenceFormCronErrorTypes.WRONG_FORMAT:
+                            setCronError(errorMessages.wrongFormat.id);
+                            break;
+                        case RecurrenceFormCronErrorTypes.FREQUENCY_TOO_HIGH:
+                            setCronError(errorMessages.frequencyTooHigh.id);
+                            break;
+                    }
+                } else {
+                    setCronError(null);
+                }
+            }
+            onCronValueChange(expression, !validationResult);
+        },
+        [cronError, validate, onCronValueChange],
+    );
+
+    const handleOnBlur = useCallback(
+        (value: string) => {
+            const validationResult = validate(value);
+            // NESTOR
```

Comment:
forgotten comment and next if+switch are redundant with handler above

Reply:
Fixed.